### PR TITLE
Move files from gamelab/ to spritelab/ or p5lab/

### PR DESCRIPTION
**Pull request by ajpal**
Part 1: #29917
Part 2: #29918
Part 3: #29925

See [doc](https://docs.google.com/document/d/1xJb79MH0Ga1pypFNyttVeNfmHm-tBWNyjGngf2dRfWs/) for more details
This moves common files out of gamelab/ into p5lab/ and Sprite Lab specific files into spritelab/

### CODE REVIEW

**Comment by jmkulwik** [comment]
Wow. This diff is great!!
<- OLD | NEW->
p5oldnew2

File: apps/src/code-studio/components/ShareAllowedDialog.jsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ import AbuseError from './AbuseError';
 import SendToPhone from './SendToPhone';
 import color from '../../util/color';
 import * as applabConstants from '../../applab/constants';
-import * as gamelabConstants from '@cdo/apps/p5lab/gamelab/constants';
+import * as gamelabConstants from '@cdo/apps/p5lab/constants';
```
Comment (jmkulwik) [inline apps/src/code-studio/components/ShareAllowedDialog.jsx:13]: Worth changing to `import * as p5labConstants` at this point?

Comment (jmkulwik) [reply apps/src/code-studio/components/ShareAllowedDialog.jsx]: (Also a general question about this PR. I'm guessing you're planning to leave renaming variables/filenames for later)

File: apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx [hunk 2]
```diff
@@ -5,8 +5,8 @@ import Radium from 'radium';
 import color from '@cdo/apps/util/color';
 import {AnimationCategories, CostumeCategories} from '../constants';
 var msg = require('@cdo/gamelab/locale') || require('@cdo/spritelab/locale');
-import animationLibrary from '../animationLibrary.json';
-import spriteCostumeLibrary from '../spriteCostumeLibrary.json';
+import animationLibrary from '../gamelab/animationLibrary.json';
+import spriteCostumeLibrary from '../spritelab/spriteCostumeLibrary.json';
```
Comment (jmkulwik) [inline apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx:9]: Are you keeping a list of next TODOs? Removing this dependency feels like a good one for the list of next steps.

File: apps/src/p5lab/P5Lab.js [hunk 3]
```diff
@@ -26,9 +26,9 @@ var dropletConfig = require('./gamelab/dropletConfig');
 var JSInterpreter = require('@cdo/apps/lib/tools/jsinterpreter/JSInterpreter');
 import * as apiTimeoutList from '@cdo/apps/lib/util/timeoutList';
 var JsInterpreterLogger = require('@cdo/apps/JsInterpreterLogger');
-var GameLabP5 = require('./gamelab/GameLabP5');
-var gameLabSprite = require('./gamelab/GameLabSprite');
-var gameLabGroup = require('./gamelab/GameLabGroup');
+var GameLabP5 = require('./GameLabP5');
+var gameLabSprite = require('./GameLabSprite');
+var gameLabGroup = require('./GameLabGroup');
```
Comment (jmkulwik) [inline apps/src/p5lab/P5Lab.js:31]: 🎉

File: apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx (hunk 2 above)
Comment (jmkulwik) [reply apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx:9]: (and similar dependencies in other places)

File: apps/src/code-studio/components/ShareAllowedDialog.jsx (hunk 1 above)
Comment (jmkulwik) [reply apps/src/code-studio/components/ShareAllowedDialog.jsx:13]: (and renaming other similar variables/filenames in other places). Nevermind about doing it right now - that would be a huge diff 😄 I definitely understand why you didn't do it as part of this change.

File: apps/src/p5lab/spritelab/SpriteLab.js [hunk 4]
```diff
@@ -3,7 +3,7 @@ import {commands} from './commands';
 import * as spriteUtils from './spriteUtils';
 import Sounds from '@cdo/apps/Sounds';
 import {getStore} from '@cdo/apps/redux';
-import {clearConsole} from '../gamelab/textConsoleModule';
+import {clearConsole} from './textConsoleModule';
```
Comment (jmkulwik) [inline apps/src/p5lab/spritelab/SpriteLab.js:6]: 💯

**Review by jmkulwik — APPROVED** [review]
This is awesome!!

File: apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx (hunk 2 above)
Comment (ajpal) [reply apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx:9]: Tracking next steps in the doc for now. Fleshing that out and adding JIRA tickets is up next for me.
+1, refactoring this to not import the json files for both labs seems like a good thing to do.

**Pull request merged**